MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -1271,17 +1275,15 @@ public final class MagazineLayout: UICollectionViewLayout {
     bottomInset: CGFloat)
     -> TargetContentOffsetAnchor?
   {
+    let insetBounds = bounds.inset(by: .init(top: topInset, left: 0, bottom: bottomInset, right: 0))
     var visibleItemLocationFramePairs = [ElementLocationFramePair]()
-    for itemLocationFramePair in modelState.itemLocationFramePairs(forItemsIn: bounds) {
+    for itemLocationFramePair in modelState.itemLocationFramePairs(forItemsIn: insetBounds) {
       visibleItemLocationFramePairs.append(itemLocationFramePair)
     }
+    visibleItemLocationFramePairs.sort { $0.elementLocation < $1.elementLocation }
```

Comment:
there were some bugs because this array wasn't guaranteed to be sorted before - whoops!

Reply:
good find!